### PR TITLE
feat: detect and request permissions in the popup page, close #1125

### LINKED ISSUE #1125: [Demand] Add a request permission button in the popup page if Maskbook has no access to the current opening domain
## Metadata

Field | Value
--- | ---
Previous Stage | ...
Figma Link | Figma_Link_Here____
Expected UX prototype due | 2020-W??-5
Actual UX prototype release | -
Expected engineering due | 2020-W??-5
Corresponding Pull Request | -
Completion Version | -

## Catalogue

- ...
- ...
- ...

### PR DESCRIPTION
**Pull request by Jack-Works**

### CODE REVIEW

File: src/_locales/zh/messages.json [hunk 1]
```diff
@@ -139,6 +139,8 @@
     "person_or_group_in_list_many": "{{people}} 等，共 {{count}} 人",
     "person_or_group_in_list_many_no_preview": "共 {{count}} 人",
     "personas": "角色",
+    "popup_missing_permission": "",
```
Comment (Tedko) [inline src/_locales/zh/messages.json:142]: "Maskbook 缺少一些必須的權限"
"修復"

File: src/extension/popup-page/UI.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            display: 'none',
+        },
+    },
+    header: {
+        margin: theme.spacing(2, 0),
+        '&:first-child': {
+            marginTop: 0,
+        },
+    },
+    footer: {},
```
Comment (septs) [inline src/extension/popup-page/UI.tsx:54]: why?

File: src/social-network/defaults/emptyDefinition.ts [hunk 3]
```diff
@@ -13,7 +13,8 @@ import { noop } from 'lodash-es'
 export const emptyDefinition: SocialNetworkUIDefinition = {
     acceptablePayload: ['latest'],
     friendlyName: '',
-    requestPermission: () => Promise.resolve(true),
+    hasPermission: async () => true,
```
Comment (septs) [inline src/social-network/defaults/emptyDefinition.ts:16]: why with `async`?

**Review by septs — APPROVED** [review]

File: src/social-network/defaults/emptyDefinition.ts (hunk 3 above)
Comment (Jack-Works) [reply src/social-network/defaults/emptyDefinition.ts:16]: they're the same

**Pull request merged**